frontend/src/main/scala/bloop/engine/tasks/compilation/CompileGraph.scala

```diff
@@ -197,7 +183,7 @@ object CompileGraph {
                 case a: ReporterAction.ProcessEndCompilation =>
                   a.code match {
                     case BspStatusCode.Cancelled | BspStatusCode.Error =>
-                      reporter.processEndCompilation(previousSuccessfulProblems, a.code, None, None)
+                      reporter.processEndCompilation(previousProblems, a.code, None, None)
                       reporter.reportEndCompilation()
```

Comment:
Changes here (and similar below) are not strictly related to best effort compilation, but I found out that clicking go-to-reference from metals on a dependency project pointing to a dependent project, I would get the "Deduplicating compilation of (...)" log and I would lose all of the diagnostics from metals. This (and the same below) is the culprit, and it seems to me that returning previousProblems (not necessarily from last successful run) is more correct here

Reply:
och, good catch! Any chance to get it as a separate fix maybe?

Reply:
Sure!